src/lib.rs

```diff
@@ -1460,7 +1518,33 @@ pub trait Itertools: Iterator {
         Self: Sized,
         Self::Item: Clone + Eq + Hash,
     {
-        unique_impl::unique(self)
+        unique_impl::unique_with_hasher(self, RandomState::new())
+    }
+
+    /// Return an iterator which yields the same elements as the one returned by
+    /// [.unique()](crate::Itertools::unique), but uses the specified hash builder to hash the
+    /// elements for comparison.
+    ///
+    /// Warning: `hash_builder` is normally randomly generated, and is designed to allow it's
+    /// users to be resistant to attacks that cause many collisions and very poor performance.
+    /// Setting it manually using this function can expose a DoS attack vector.
+    ///
+    /// ```
+    /// use std::hash::RandomState;
```

Comment:
What do you think about using a 3rd party hasher here in the docs? This would actually help ensure the types work with other hashers.

For example, if the (defaulted) generic argument is forgotten in one of the `impl` blocks:
``` rust
impl<I, Key, F> Iterator for DuplicatesBy<I, Key, F> { }
// which actually means
impl<I, Key, F> Iterator for DuplicatesBy<I, Key, F, std::hash::RandomState> { }
```

the types only are iterators with the default hasher.

But I didn't think it was worth pulling in a dependency just for this.

Reply:
Is it feasible to test all of them with a simple Hasher that always hashes to 0? Could be written by ourself, and is maybe feasible to run in each test run.

(I’m on mobile and my phone is too slow to display the diffs, so please excuse if this is already addressed.)